app/jobs/cancel_hold_job.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class CancelHoldJob < ApplicationJob
+  queue_as :default
+
+  def perform(hold_key:, session_token:)
+    symphony_client = SymphonyClient.new
+    redis_client = Redis.new
+
+    response = symphony_client.cancel_hold(hold_key: hold_key, session_token: session_token)
+
+    case response.status
+    when 200
+      redis_client.set("cancel_hold_#{hold_key}", {
+        hold_id: hold_key,
+        result: :success
+      }.to_json)
+    else
+      error_message_raw = JSON.parse response.body
+      error_message = error_message_raw&.dig('messageList')&.first&.dig('message') || 'Something went wrong'
```

Comment:
@cdmo why we need a default here? just curios if the response does not always include a message?

Reply:
for weird cases like when we get bad responses from Sirsi?

Reply:
Nice catch, I wasn't sure tbh. It was more of a safety than anything.

We're probably going to have to write a small service class or model or something to handle Sirsi error translations I think